### PR TITLE
Parse database

### PR DESCRIPTION
**Pull request by emokabe**
### Context
This functionality doesn’t not build from a past PR, but uses one of the tabs in the tab bar controller implemented at the beginning of the project. When the second tab from the left is pressed, the app displays a list of courses stored in the Parse database.

### Test plan
1. Run the simulator, then click the login button to login with Facebook credentials.
2. After seeing the Home Screen with posts, press on the second tab from the left.
3. The user should see a list of courses that the user is enrolled in.
4. Clicking the leftmost tab will bring the user back to the home question feed.

### MVP features implemented
This PR is the first part of the “Select a course from a list to view its feed” feature. While the list of courses is displayed, I have yet to add the course selection functionality.

https://user-images.githubusercontent.com/107252243/179637160-f4152bba-69cc-43b7-891a-7e47cce39aee.mov

### CODE REVIEW

File: CapstoneProject/AppDelegate.m [hunk 1]
```diff
@@ -21,6 +22,16 @@ - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(
     // Override point for customization after application launch.
     [FBSDKLoginButton class];
     [FBSDKApplicationDelegate.sharedInstance application:application didFinishLaunchingWithOptions:launchOptions];
+    
+    ParseClientConfiguration *config = [ParseClientConfiguration  configurationWithBlock:^(id<ParseMutableClientConfiguration> configuration) {
+        
+        configuration.applicationId = @"Ah8NtLvAUjjjk6zIro3QeC8S3g3bYmfSQT2CCuwX";
+        configuration.clientKey = @"WH4s0Z5NL8Aa2BxC202b1eqwLgtVAOBCnTpEF9u4";
```
Comment (blai) [inline CapstoneProject/AppDelegate.m:29]: I wonder if CodePath has any recommendation for secrets, normally we don't commit them to a publicly visible repo.

File: CapstoneProject/AppDelegate.m [hunk 2]
```diff
@@ -21,6 +22,16 @@ - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(
     // Override point for customization after application launch.
     [FBSDKLoginButton class];
     [FBSDKApplicationDelegate.sharedInstance application:application didFinishLaunchingWithOptions:launchOptions];
+    
+    ParseClientConfiguration *config = [ParseClientConfiguration  configurationWithBlock:^(id<ParseMutableClientConfiguration> configuration) {
+        
+        configuration.applicationId = @"Ah8NtLvAUjjjk6zIro3QeC8S3g3bYmfSQT2CCuwX";
```
Comment (blai) [inline CapstoneProject/AppDelegate.m:28]: Usually these should be a constant value defined somewhere, or something we load from a config.

File: CapstoneProject/AppDelegate.m (hunk 1 above)
Comment (emokabe) [reply CapstoneProject/AppDelegate.m:29]: I actually did this the same way that CodePath taught us — they told us to put all of this in AppDeledate.m ([see this file from the third project](https://github.com/emokabe/Instagram/blob/main/Instagram/AppDelegate.m)). I can see why this isn't a good idea though.

File: CapstoneProject/View Controllers/SelectCourseViewController.m [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+            self.courseArray = [NSMutableArray arrayWithArray:courses];
+            [self.tableView reloadData];
+        }
+    }];
+}
+
+
+- (void)getCoursesWithCompletion:(void(^)(NSArray *courses, NSError *error))completion {
+    PFQuery *query = [PFQuery queryWithClassName:@"Course"];
+    query.limit = 20;
```
Comment (blai) [inline CapstoneProject/View Controllers/SelectCourseViewController.m:42]: Not very familiar with Parse, does this limit means the page size (e.g. if we run out of all 20 items, we will load another batch of 20), or are we just making the list cap at 20 (e.g. hard limit)?

**Comment by blai** [comment]
LGTM. For the pod dependencies, do we have a way to exclude them from being part of commits?

File: CapstoneProject/View Controllers/SelectCourseViewController.m (hunk 3 above)
Comment (emokabe) [reply CapstoneProject/View Controllers/SelectCourseViewController.m:42]: It means that the list caps at 20. Now that you pointed it out, I realize that this wasn't necessary — I'll remove this for now and likely replace it with loading another batch whenever the user runs out.

**Pull request merged**